Make stream create/delete operation asynchronous

igkon opened the pull request:
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Igor <hokletka@gmail.com>

  Make create and delete stream operations asynchronous with CompleatableFuture.handle().
  Now we try to delete the stream whether it is sealed or not
  Add two methods to complete create or delete operation for stream.

Commit:
Make stream create/delete operation asynchronous

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Igor <hokletka@gmail.com>

  Make create and delete stream operations asynchronous with CompleatableFuture.handle().
  Now we try to delete the stream whether it is sealed or not
  Add two methods to complete create or delete operation for stream.

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:
```diff
@@ -596,31 +605,38 @@ void submitStreamReadOperation(final PathOperation streamOp, final String nodeAd
 		// TODO: Alex, issue SDP-51
 	}
 
+	boolean completeStreamDeleteOperation(final PathOperation streamOp, final boolean result, final Throwable thrown,
```
Couldn't we get the scope/stream names from the `streamOp`? Just to not to pass extra arguments

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java:
```diff
@@ -554,6 +554,23 @@ void submitStreamOperation(final PathOperation streamOp, final OpType opType) {
 		}
 	}
 
+	boolean completeStreamCreateOperation(final PathOperation streamOp, final boolean result, final Throwable thrown,
```
Couldn't we get the scope/stream names from the `streamOp`? Just to not to pass extra arguments

Commit:
Fix of completion methods

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Igor <hokletka@gmail.com>
